src/Adapter/Amp/EventLoop.php

```diff
@@ -13,7 +13,7 @@ class EventLoop implements \M6Web\Tornado\EventLoop
     public function wait(Promise $promise)
     {
         try {
-            return \Amp\Promise\wait(Internal\PromiseWrapper::downcast($promise)->getAmpPromise());
+            return \Amp\Promise\wait(Internal\PromiseWrapper::downcast($promise)->markYielded()->getAmpPromise());
```

Comment:
I'm not very comfortable with `markYielded` function, because it's too easy to misunderstood its usage and to create tricky bugs… In this case, we could add a function `Internal\PromiseWrapper::wait`, marking the promise and forwarding the call to `Amp`.